src/components/molecules/Popover/Popover.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { ReactNode, useRef } from "react";
+
+import { useClickOutside } from "hooks/useClickOutside";
+import { useShowHide } from "hooks/useShowHide";
+
+import "./Popover.scss";
+
+type PopoverProps = {
+  overlay: ReactNode;
+  defaultShow?: boolean;
+  closeRoot?: boolean;
+  className?: string;
+};
+
+export const Popover: React.FC<PopoverProps> = ({
```

Comment:
This looks like having solved enough issues that we can use the knowledge to bootstrap (pardon the pun) our own modal component